addons/mail/static/src/new/messaging_menu/messaging_menu.js

```diff
@@ -45,4 +45,14 @@ export class MessagingMenu extends Component {
         // without dropdownitem...
         document.body.click();
     }
+
+    getMenuCounter() {
+        const menuCounter =
+            this.messaging.state.discuss.inbox.counter +
+            Object.values(this.messaging.state.threads).filter((thread) => thread.is_pinned)
+                .length +
+            Object.keys(this.messaging.state.notificationGroups).length +
+            this.messaging.state.isNotificationPermissionDefault;
+        return menuCounter;
+    }
```

Comment:
I don't see any cleanup of `notificationGroups` from read. So I assume marking as read does not decrement the counter (without taking page refresh into account)